### PR TITLE
Fix TFTransfoXLLMHeadModel outputs

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Fix the outputs of `TFTransfoXLLMHeadModel` (in the case without `labels`) - current TF returns `softmax_output` while PT returns `prediction_scores`:

- Current PT

    https://github.com/huggingface/transformers/blob/6f9d8dc1567889b073b91059c9b19e9a6813abfa/src/transformers/models/transfo_xl/modeling_transfo_xl.py#L1119
    and

    https://github.com/huggingface/transformers/blob/6f9d8dc1567889b073b91059c9b19e9a6813abfa/src/transformers/models/transfo_xl/modeling_transfo_xl.py#L1138-L1140

- Current TF

    https://github.com/huggingface/transformers/blob/6f9d8dc1567889b073b91059c9b19e9a6813abfa/src/transformers/models/transfo_xl/modeling_tf_transfo_xl.py#L1005-L1006

## Remarks:

  - The case with `labels` is much more complicated - to be addressed in the future.
  - The current PT/TF equivalence test has a bit flaw and doesn't detect this issue. A WIP PR #16557 is on its way (with other enhancements)!

### CODE REVIEW

File: src/transformers/models/transfo_xl/modeling_tf_transfo_xl.py [hunk 1]
```diff
@@ -998,12 +998,13 @@ def call(
         pred_hid = last_hidden[:, -tgt_len:]
 
         softmax_output = self.crit(pred_hid, labels, training=training)
+        prediction_scores = softmax_output if labels is None else ()
```
Comment (sgugger) [inline src/transformers/models/transfo_xl/modeling_tf_transfo_xl.py:1001]: An empty tuple? Is that really what's expected? Seems weird.

Comment (ydshieh) [reply src/transformers/models/transfo_xl/modeling_tf_transfo_xl.py:1001]: This is done in PyTorch version. I am not sure about the reason why not to output `prediction_scores` (empty tuple).

`softmax_output` is returned by [ProjectedAdaptiveLogSoftmax](https://github.com/huggingface/transformers/blob/6f9d8dc1567889b073b91059c9b19e9a6813abfa/src/transformers/models/transfo_xl/modeling_transfo_xl_utilities.py#L29), which is quite complicated. In particular, when `labels` is passed to `ProjectedAdaptiveLogSoftmax`, the logic is very different from not passing `labels`.

Comment (ydshieh) [reply src/transformers/models/transfo_xl/modeling_tf_transfo_xl.py:1001]: Oh, there is one reason doing so in PyTorch version:

```
 softmax_output.view(bsz, tgt_len, -1)
```
This shape only works when `labels` is not used.

**Review by sgugger — APPROVED** [review]
Ok to align the two APIs!

**Review by gante — APPROVED** [review]
LGTM, as it makes the two versions closer to each other

**Review by patrickvonplaten — APPROVED** [review]
